Source/Knowledge/CompKnowledge.cs

```diff
@@ -238,30 +238,48 @@ public void AcquireWeaponKnowledge(FactionDef faction)
-                        stringBuilder.Append(pawn.gender.GetPronoun().CapitalizeFirst() + " is a " + role[0] + " and gets extra " + role[1] + " weapons from ");
+                        var weapons = tech.UnlockedWeapons().Where(x => TestIfWeapon(x, isFighter));
+                        if (weapons.Any()) foreach (ThingDef w in weapons) proficientWeapons.Add(w);
+                        if (Prefs.LogVerbose) stringBuilder.Append(pawn.gender.GetPossessive().ToLower() + " faction's tech level");
                     }
-                    if (ModBaseHumanResources.WeaponPoolIncludesTechLevel)
-                    {
-                        foreach (ResearchProjectDef tech in DefDatabase<ResearchProjectDef>.AllDefs.Where(x => TechPool(isPlayer, x, startingTechLevel, faction)))
+                }
+                if (isPlayer)
+                {
+                    if (ModBaseHumanResources.unlocked.knowAllStartingWeapons) {
+                        proficientWeapons.AddRange(ModBaseHumanResources.unlocked.startingWeapons);
+                        if (Prefs.LogVerbose)
                         {
-                            var weapons = tech.UnlockedWeapons().Where(x => TestIfWeapon(x, isFighter));
-                            if (weapons.Any()) foreach (ThingDef w in weapons) proficientWeapons.Add(w);
+                            if (ModBaseHumanResources.WeaponPoolIncludesTechLevel && (isFighter || isShooter))
+                            {
+                                stringBuilder.Append(" and the starting scenario.");
+                            }
+                            else
+                            {
+                                stringBuilder.Append(pawn.gender.GetPronoun().CapitalizeFirst() + " gets all weapons from the starting Scenario.");
+                            }
                         }
-                        if (Prefs.LogVerbose) stringBuilder.Append(pawn.gender.GetPossessive().ToLower() + " faction's tech level");
                     }
-                    if (isPlayer && ModBaseHumanResources.WeaponPoolIncludesScenario)
+                    else if (ModBaseHumanResources.WeaponPoolIncludesScenario && (isFighter || isShooter))
                     {
                         proficientWeapons.AddRange(ModBaseHumanResources.unlocked.startingWeapons.Where(x => TestIfWeapon(x, isFighter)));
-                        string connector = ModBaseHumanResources.WeaponPoolIncludesTechLevel ? " and " : "";
-                        if (Prefs.LogVerbose) stringBuilder.Append(connector + "the starting scenario");
+                        if (Prefs.LogVerbose)
+                        {
+                            string connector = ModBaseHumanResources.WeaponPoolIncludesTechLevel ? " and " : "";
+                            stringBuilder.Append(connector + "the starting scenario.");
+                        }
                     }
-                    if(Prefs.LogVerbose) stringBuilder.Append(".");
```

Comment:
Changes to the string builder made this very convoluted! What's the angle here? I feel it would be simpler to add a third "role"

Reply:
You are right, I will change it.